xds/internal/xdsclient/transport/ads/ads_stream.go

```diff
@@ -625,17 +630,28 @@ func (s *Stream) onError(err error, msgReceived bool) {
 			rs.State = ResourceWatchStateStarted
 		}
 	}
+	s.mu.Unlock()
 
-	// Forward the error to the channel.
+	// Note that we do not consider it an error if the ADS stream was closed
```

Comment:
I still don't get what not considering it an error means, as per my original comment. I see it unconditionally reporting OnADSStreamError, so does trigger some logic at a higher layer? If so can you specify.

Reply:
The error is simply logged by the `authority`. In the existing code, this is handled here: https://github.com/grpc/grpc-go/blob/ad81c20503be8c36d929741078e1a53a292e4048/xds/internal/xdsclient/authority.go#L451

In this new code, this will be handled here: 
https://github.com/easwars/grpc-go/blob/0bf5608bf581a01ac86881eac7837f2dbde0b344/xds/internal/xdsclient/authority.go#L173

This logic is mentioned in this gRFC here: https://github.com/grpc/proposal/blob/master/A57-xds-client-failure-mode-behavior.md#handling-ads-stream-termination

Reply:
Ah ok thanks.